### DISCUSSION TITLE
Hybrid HTTP / WS transports

### DISCUSSION THREAD

**Discussion by AadamZ5** [n1ru4l/graphql-live-query#746] [Q&A] [answered]
One of the solutions I have been hoping for in GraphQL is the ability to just "watch" my data. Subscriptions would work, but something didn't feel right about writing a whole new operation on my GraphQL endpoint just to instead watch a certain type of data. The solution I imagine now would be to write a subscription operation like `watchUser(...)` that would return similar data to my query `getUser(...)` operation. They both return the same thing, one of them just streams. I don't want to have this duplication in my endpoint. Instead, I'd love to just have a `user(...)` query operation that can be streamed when `@live` is included.

However, I will be eventually deploying my end-solution to the cloud. Having each and every client use `graphql-ws` for every type of operation will not be feasible, since that means every client will need to maintain a websocket connection to the cloud instance. Correct me if I'm wrong, but I think `graphql-ws` needs to keep a connection, even if no data is currently being transferred. 

If what I said above is correct, then this leads me to the next solution I was hoping for. For context, it's been possible to specify a different subscriptions endpoint, that of which commonly uses some sort of streaming connection, typically websockets. Then, the typical single-response operations in `Query` and `Mutation` use a simple HTTP request. This ability to use websockets only when necessary is very useful, and keeps internet usage low when it can be. 

I haven't yet taken the time to inspect the implementation details of this library (There's so much to the GQL ecosystem!) but I was hoping that there would be some way to change connections based on if the `@live` decorator is supplied, or not. I have to imagine that this involves some client-side configuration as well, but it would be well worth it. I really don't want to have my entire userbase need to keep a websocket connection when accessing our web app. 

If you can't take time to explain a sample implementation, I understand. If at least you could explain briefly why or why not that would be possible, I would greatly appreciate it.

**Comment by n1ru4l** [comment, marked as answer]
In your GraphQL network layer, you can use the [`isLiveQueryOperationDefinitionNode`](https://github.com/n1ru4l/graphql-live-query/tree/main/packages/graphql-live-query#islivequeryoperationdefinitionnode) function for identifying whether the operation is a live query and then use the corresponding transport you want to use.

E.g. if you are using apollo client you could use the [split link for this](https://www.apollographql.com/docs/react/data/subscriptions/#3-split-communication-by-operation-recommended).

**Reply by AadamZ5** [reply]
Perfect! I haven't explored too much client side yet, especially the network layer aspects. I'll check this out.
